applicationset/generators/duck_type.go

```diff
@@ -19,24 +19,27 @@ import (
 
 	"github.com/argoproj/argo-cd/v3/applicationset/utils"
 	argoprojiov1alpha1 "github.com/argoproj/argo-cd/v3/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/argo-cd/v3/util/settings"
 )
 
 var _ Generator = (*DuckTypeGenerator)(nil)
 
 // DuckTypeGenerator generates Applications for some or all clusters registered with ArgoCD.
 type DuckTypeGenerator struct {
-	ctx       context.Context
-	dynClient dynamic.Interface
-	clientset kubernetes.Interface
-	namespace string // namespace is the Argo CD namespace
+	ctx             context.Context
+	dynClient       dynamic.Interface
+	clientset       kubernetes.Interface
+	namespace       string // namespace is the Argo CD namespace
+	clusterInformer *settings.ClusterInformer
 }
 
-func NewDuckTypeGenerator(ctx context.Context, dynClient dynamic.Interface, clientset kubernetes.Interface, namespace string) Generator {
+func NewDuckTypeGenerator(ctx context.Context, dynClient dynamic.Interface, clientset kubernetes.Interface, namespace string, clusterInformer *settings.ClusterInformer) Generator {
```

Comment:
Can the namespace be different than the clusterInformer namespace?

Reply:
[Nope](https://github.com/argoproj/argo-cd/blob/abb354b5e32e82cf1fc13daa9cb4818f9630a5b8/applicationset/generators/duck_type.go#L31), its the [same namespace](https://github.com/argoproj/argo-cd/blob/abb354b5e32e82cf1fc13daa9cb4818f9630a5b8/cmd/argocd-applicationset-controller/commands/applicationset_controller.go#L96) that gets [passed to the generators](https://github.com/argoproj/argo-cd/blob/abb354b5e32e82cf1fc13daa9cb4818f9630a5b8/cmd/argocd-applicationset-controller/commands/applicationset_controller.go#L215).